iconservice/iconscore/icon_pre_validator.py

```diff
@@ -77,10 +81,23 @@ def execute_to_check_out_of_balance(
         else:
             self._check_from_can_charge_fee_v3(params, step_price)
 
-    def _check_data_size(self, params: dict):
+    @staticmethod
+    def _check_message_data(data):
+        """
+        Check if the message data is a lowercase hex string
+        :param data: input data of message type
+        """
+
+        if not isinstance(data, str) or \
+                data[:2] != '0x' or \
+                not is_lowercase_hex_string(data[2:]):
```

Comment:
```suggestion
                (not is_lowercase_hex_string(data[2:]) and len(data) % 2 == 0)
```
0x123 (X)
0x0123 (O)